docs: misc improved docs, including rust watcher example (#132)
ActivityWatch/docs: src/examples/writing-watchers.rst

> Writing your first watcher in python

Reviewer: ```suggestion
Writing your first watcher in Python
```

.. literalinclude:: client.py

> Writing your first watcher in rust

Reviewer: ```suggestion
Writing your first watcher in Rust
```

==================================

> To get started with writing watchers in Rust, you need to add the `aw-client-rust` and `aw-model` crates to your `Cargo.toml` file.

Reviewer: ```suggestion
To get started with writing watchers in Rust, you need to add the ``aw-client-rust`` and ``aw-model`` crates to your ``Cargo.toml`` file.
```

> The most upto date versions depend directly on :gh-aw:`aw-server-rust`.

Reviewer: ```suggestion
The most up-to-date versions depend directly on :gh-aw:`aw-server-rust`.
```
